lib/types/operation.js

```diff
@@ -252,6 +258,36 @@ Operation.prototype.getType = function (param) {
   }
 };
 
+/**
+ * adds an inline schema (model) to a hash, where we can ref it later
+ * @param {object} schema a schema
+ * @return {number} the ID of the schema being added, or null
+ **/
+Operation.prototype.addInlineModel = function (schema) {
+  var len = this.inlineModels.length;
+  var model = this.resolveModel(schema, {});
+  if(model) {
+    this.inlineModels.push(model);
+    return len; // to be added to the ID...eg: 'inline#123', where 123 is the ID
+  }
+  return null; // report errors?
+};
+
+/**
+ * gets the internal ref to an inline model
+ * @param {string} inline_str a string reference to an inline model
+ * @return {Model} the model being referenced. Or null
+ **/
+Operation.prototype.getInlineModel = function(inline_str) {
```

Comment:
Abstraction, this is where we cache the model for de-referencing. I wouldn't want that logic inside of operation#getType... And in the commit below it's used to return the full string ID, further abstracting things.